src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,1638 @@
+            nn.ReLU(inplace=True),
+        )
+
+
+class FPNLayer(nn.Module):
+    def __init__(self, in_features: int, lateral_features: int):
+        """A Feature Pyramid Network Layer. It creates a feature map by aggregating features from the previous and backbone layer.
+        Due to the spartial mismatch, the tensor coming from the previous layer is upsample.
+
+        Args:
+            in_features (int): The number of input features (channels)
+            lateral_features (int): The number of lateral features (channels)
+        """
+        super().__init__()
+        self.proj = nn.Sequential(
+            nn.Conv2d(lateral_features, in_features, kernel_size=1, padding=0, bias=False),
+            nn.GroupNorm(32, in_features),
+        )
+
+        self.block = ConvLayer(in_features, in_features)
+
+    def forward(self, down: Tensor, left: Tensor) -> Tensor:
+        left = self.proj(left)
+        down = F.interpolate(down, size=left.shape[-2:], mode="nearest")
+        down += left
+        down = self.block(down)
+        return down
+
+
+class FPNModel(nn.Module):
+    def __init__(self, in_features: int, lateral_widths: List[int], feature_size: int = 256):
+        """Feature Pyramid Network, given an input tensor and a set of features map of different feature/spatial size, it creates a list of features map with different the same feature size.
+
+        Args:
+            in_features (int): The number of input features (channels)
+            lateral_widths (List[int]): A list with the features (channels) size of each lateral connection
+            feature_size (int, optional): The features (channels) of the resulting feature maps. Defaults to 256.
+        """
+        super().__init__()
+        self.stem = ConvLayer(in_features, feature_size)
+        self.layers = nn.Sequential(*[FPNLayer(feature_size, lateral_width) for lateral_width in lateral_widths[::-1]])
+
+    def forward(self, features: List[Tensor]) -> List[Tensor]:
+        fpn_features: List[Tensor] = []
+        last_feature: Tensor = features.pop()
+        x: Tensor = self.stem(last_feature)
+        for layer, left in zip(self.layers, features[::-1]):
+            x = layer(x, left)
+            fpn_features.append(x)
+        return fpn_features
+
+
+class MaskFormerPixelDecoder(nn.Module):
+    def __init__(self, *args, feature_size: int = 256, mask_feature_size: int = 256, **kwargs):
+        """Pixel Decoder Module proposed in `Per-Pixel Classification is Not All You Need for Semantic Segmentation <https://arxiv.org/abs/2107.06278>`_. It first run the backbone's feature into a Feature Pyramid Network creating a list of features map. Then, it projects the last one to the correct `mask_size`
+
+        Args:
+            feature_size (int, optional): The features (channels) of FPN feature maps. Defaults to 256.
+            mask_feature_size (int, optional): The features (channels) of the target masks size $C_{\epsilon}$ in the paper. Defaults to 256.
```

Comment:
See any other model for the standard way we describe and type args with defaults.
Special behavior: we can't use `$ $` for inline math so the syntax is `\\( \\)`